### PR TITLE
chore: Update QUERY_LOGGER and SQL_QUERY_MUTATOR signatures

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

This PR addresses Superset 3.0 task [#45](https://github.com/orgs/apache/projects/201/views/1?pane=issue&itemId=28098933).

### TESTING INSTRUCTIONS

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: UPDATING.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ assists people when migrating to a new version.
 
 ## Next
 
+- [24029](https://github.com/apache/superset/pull/24029) Removes the `user` and `username` arguments for the QUERY_LOGGER and SQL_QUERY_MUTATOR methods respectively. If the username for the current user is required the `superset.utils.core.get_username` method should be used.
```
Comment (michael-s-molina) [inline UPDATING.md:27]: Could you move this to the **Breaking Changes** section?

**Review by michael-s-molina — APPROVED** [review]
Thank you for the cleanup @john-bodley. I left a comment but other then that LGTM.

**Review by michael-s-molina — APPROVED** [review]

**Comment by eschutho** [comment]
@john-bodley I'm just tagging this with risk:breaking-change for anyone that might be overriding the config.

**Comment by eschutho** [comment]
@michael-s-molina should we put something in the UPDATING file for 3.0 on this, too? This broke our integration, so maybe having the change in updating could help other people upgrading as well b/c they will have to import the user into their config overwrite from global now.

**Comment by michael-s-molina** [comment]
> @michael-s-molina should we put something in the UPDATING file for 3.0 on this, too? This broke our integration, so maybe having the change in updating could help other people upgrading as well b/c they will have to import the user into their config overwrite from global now.

Definitely. I think there was a miscommunication in the review because there's breaking change text added in this PR but it seems it's not related to the particular change.

**Comment by michael-s-molina** [comment]
> > @michael-s-molina should we put something in the UPDATING file for 3.0 on this, too? This broke our integration, so maybe having the change in updating could help other people upgrading as well b/c they will have to import the user into their config overwrite from global now.
> 
> Definitely. I think there was a miscommunication in the review because there's breaking change text added in this PR but it seems it's not related to the particular change.

@eschutho @john-bodley I added the breaking change text [here](https://github.com/apache/superset/pull/24295).

**Pull request merged**